### PR TITLE
[#151023459] configure CircleCI

### PR DESCRIPTION
**Pull request by MothOnMars**
- add rspec_junit_formatter gem
- remove deprecated codeclimate-test-reporter gem

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    working_directory: ~/asis
+
+    steps:
+      - checkout
+
+      - run:
+          name: Install Elasticsearch
+          command: |
+            curl -O https://download.elasticsearch.org/elasticsearch/elasticsearch/elasticsearch-1.7.2.deb && sudo dpkg -i elasticsearch-1.7.2.deb && true
```
Comment (noremmie) [inline .circleci/config.yml]: Just curious.. why not version 1.7.3? I guess there are no significant differences between 1.7.2 and 1.7.3 w.r.t. asis.

Comment (MothOnMars) [reply .circleci/config.yml]: I used 1.7.2 because it's what we use for Travis: https://github.com/GSA/asis/blob/master/.travis.yml#L6

Comment (noremmie) [reply .circleci/config.yml]: Got it. Thanks for indulging my curiosity.

Comment (MothOnMars) [reply .circleci/config.yml]: NP. I'm sure I could use 1.7.3, but the CircleCI configs have been such a pain in the *** so far, that I tried to make as few deviations from our Travis setup, etc. as possible. I'm inclined to change it if/when we change our ES version for asis.

**Pull request merged**